<launch>

<arg name="gtTopic" default="GT"/>
<arg name="cameraTopic" default="/camera0/color/image_raw"/>
<arg name="jsonPath" default="$(find-pkg-share omni3d_ros)/configs/args.json"/>
<arg name="omniTopic" default="fullomni3d"/>

<node pkg="rviz2" exec="rviz2" name="rviz2" output="screen" args="-d $(find-pkg-share omni3d_ros)/rviz/mapping.rviz">
</node>

<node pkg="omni3d_ros" exec="GTNode" name="GTNode">
  <param name="csvPath" value="/SIMP/data/raw_gt.csv"/> 
 <param name="gtTopic" value="$(var gtTopic)"/> 
  <param name="jsonPath" value="$(var jsonPath)"/>
   <param name="cameraTopic" value="$(var cameraTopic)"/>
</node>

<node pkg="omni3d_ros" exec="SIMPNode" name="SIMPNode">
  <param name="omniTopic" value="$(var omniTopic)"/> 
 <param name="gtTopic" value="$(var gtTopic)"/> 
  <param name="jsonPath" value="$(var jsonPath)"/>
   <param name="dumpPath" value="/SIMP/"/>
</node>

<node pkg="omni3d_ros" exec="Omni3DMappingNode" name="Omni3DMappingNode">
  <param name="omniTopic" value="$(var omniTopic)"/> 
   <param name="cameraTopic" value="$(var cameraTopic)"/>
 <param name="modelPath" value="$(find-pkg-share omni3d_ros)/models/model_recent.pth"/> 
  <param name="jsonPath" value="$(var jsonPath)"/>
   <param name="configPath" value="$(find-pkg-share omni3d_ros)/configs/trained_Omni3D.yaml"/>
</node>


 <executable cmd="sleep 3" output="screen" />
<!--
  <executable cmd="ros2 bag play /SIMP/data/2023-01-17-19-06-59_12/" output="screen" />
-->

<node name="map_server" pkg="nav2_map_server" exec="map_server" output="screen">
 <param name="yaml_filename" value="$(find-pkg-share omni3d_ros)/configs/Map.yaml"/>
</node>
 <executable cmd="ros2 lifecycle set /map_server configure" output="screen" />
 <executable cmd="ros2 lifecycle set /map_server activate" output="screen" />



</launch>
